nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -188,10 +188,16 @@ class BtActionNode : public BT::ActionNodeBase
       // setting the status to RUNNING to notify the BT Loggers (if any)
       setStatus(BT::NodeStatus::RUNNING);
 
+      // reset the flag to send the goal or not, allowing the user the option to set it in on_tick
+      should_send_goal_ = true;
+
       // user defined callback
       on_tick();
 
-      send_new_goal();
+      if(should_send_goal_)
```

Comment:
Linting says this should be `if (should_send_goal_) {`. You can reproduce locally by running `ament_cpplint` and `ament_uncrustify`